lib/rubocop/cop/solidus/spree_default_address_deprecated.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Solidus
+      # This cop finds user.default_address or user.default_address and suggest
+      # using user.ship_address or user.default_user_ship_address instead.
+      #
+      # @example
+      #
+      #   # bad
+      #   user.default_address
+      #   user.default_user_address
+      #
+      #   # good
+      #   user.ship_address
+      #   user.default_user_ship_address
+      #
+      class SpreeDefaultAddressDeprecated < Base
+        include TargetSolidusVersion
+        minimum_solidus_version 2.11
+
+        MSG = 'user.default_address and user.default_user_address are deprecated and will be removed on Solidus 3.0. ' \
+              'Please use user.ship_address or user.default_user_ship_address instead.'
+
+        def_node_matcher :default_address?, <<~PATTERN
+          (send ... :default_address)
+        PATTERN
+
+        def_node_matcher :default_user_address?, <<~PATTERN
+          (send ... :default_user_address)
+        PATTERN
+
+        def on_send(node)
+          return unless default_address?(node) || default_user_address?(node)
+
+          add_offense(node) do |corrector|
+            corrector.replace(node, node.source.gsub('default_address', 'ship_address')) if default_address?(node)
+            corrector.replace(node, node.source.gsub('default_user_address', 'default_user_ship_address')) if default_user_address?(node)
```

Comment:
I think you should `extend AutoCorrector` since you're calling `corrector.replace`  here